### ISSUE TITLE
Unable to deserialize to object: type, KeyError: 'DayOfWeek'

### ISSUE THREAD

**Issue by gnychis** [Microsoft/azure-devops-python-api#154]
All API related calls for team settings (e.g., `get_team_settings()` and `update_team_settings()` seem to fail with `Unable to deserialize to object: type, KeyError: 'DayOfWeek'`.

Client side I am using vsts 0.1.21, and server side I am on Azure DevOps Server 2019 RC1

Here is some sample debug output:

```
DEBUG:vsts.vss_client:Route template: {project}/{team}/_apis/{area}/{resource}
DEBUG:vsts.vss_client:Api version '4.1'
DEBUG:vsts.vss_client:GET https://devops.acme.com/tfs/Team%20Inheritance%20Test/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/Platform/_apis/work/teamsettings
DEBUG:vsts.vss_client:Request content: None
DEBUG:msrest.universal_http:Configuring redirects: allow=True, max=30
DEBUG:msrest.universal_http:Configuring request: timeout=100, verify=True, cert=None
DEBUG:msrest.universal_http:Configuring proxies: ''
DEBUG:msrest.universal_http:Evaluate proxies against ENV settings: True
DEBUG:urllib3.connectionpool:https://devops.acme.com:443 "GET /tfs/Team%20Inheritance%20Test/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/Platform/_apis/work/teamsettings HTTP/1.1" 200 564
DEBUG:vsts.vss_client:Response content: b'{"backlogIteration":{"id":"00000000-0000-0000-0000-000000000000"},"bugsBehavior":"asTasks","workingDays":["monday","tuesday","wednesday","thursday","friday"],"backlogVisibilities":{"Microsoft.FeatureCategory":true,"Microsoft.EpicCategory":false,"Microsoft.RequirementCategory":true},"defaultIteration":null,"defaultIterationMacro":"@currentIteration","url":"https://devops.acme.com/tfs/Team Inheritance Test/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/b54a7d18-6edb-44b8-9771-d931faeb5c86/_apis/work/teamsettings","_links":{"self":{"href":"https://devops.soroco.com/tfs/Team Inheritance Test/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/b54a7d18-6edb-44b8-9771-d931faeb5c86/_apis/work/teamsettings"},"project":{"href":"https://devops.soroco.com/tfs/Team Inheritance Test/_apis/projects/6fc0f768-5a54-4e35-bfbd-9676efd75e5e"},"team":{"href":"https://devops.soroco.com/tfs/Team Inheritance Test/_apis/projects/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/teams/b54a7d18-6edb-44b8-9771-d931faeb5c86"},"teamIterations":{"href":"https://devops.soroco.com/tfs/Team Inheritance Test/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/b54a7d18-6edb-44b8-9771-d931faeb5c86/_apis/work/teamsettings/iterations"},"teamFieldValues":{"href":"https://devops.soroco.com/tfs/Team Inheritance Test/6fc0f768-5a54-4e35-bfbd-9676efd75e5e/b54a7d18-6edb-44b8-9771-d931faeb5c86/_apis/work/teamsettings/teamfieldvalues"}}}'
DEBUG:msrest.exceptions:Unable to deserialize to object: type, KeyError: 'DayOfWeek'
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
c:\python36\lib\site-packages\msrest\serialization.py in _deserialize(self, target_obj, data)
   1293 
-> 1294                 value = self.deserialize_data(raw_value, attr_desc['type'])
   1295                 d_attrs[attr] = value

c:\python36\lib\site-packages\msrest\serialization.py in deserialize_data(self, data, data_type)
   1446             if iter_type in self.deserialize_type:
-> 1447                 return self.deserialize_type[iter_type](data, data_type[1:-1])
   1448 

c:\python36\lib\site-packages\msrest\serialization.py in deserialize_iter(self, attr, iter_type)
   1477             ))
-> 1478         return [self.deserialize_data(a, iter_type) for a in attr]
   1479 

c:\python36\lib\site-packages\msrest\serialization.py in <listcomp>(.0)
   1477             ))
-> 1478         return [self.deserialize_data(a, iter_type) for a in attr]
   1479 

c:\python36\lib\site-packages\msrest\serialization.py in deserialize_data(self, data, data_type)
   1448 
-> 1449             obj_type = self.dependencies[data_type]
   1450             if issubclass(obj_type, Enum):

KeyError: 'DayOfWeek'

During handling of the above exception, another exception occurred:

DeserializationError                      Traceback (most recent call last)
<ipython-input-16-6b5197c3a80a> in <module>
----> 1 work_client.get_team_settings(TeamContext(project_id=project_id, team="Platform"))

c:\python36\lib\site-packages\vsts\work\v4_1\work_client.py in get_team_settings(self, team_context)
   1307                               version='4.1',
   1308                               route_values=route_values)
-> 1309         return self._deserialize('TeamSetting', response)
   1310 
   1311     def update_team_settings(self, team_settings_patch, team_context):

c:\python36\lib\site-packages\msrest\serialization.py in __call__(self, target_obj, response_data, content_type)
   1226         """
   1227         data = self._unpack_content(response_data, content_type)
-> 1228         return self._deserialize(target_obj, data)
   1229 
   1230     def _deserialize(self, target_obj, data):

c:\python36\lib\site-packages\msrest\serialization.py in _deserialize(self, target_obj, data)
   1296         except (AttributeError, TypeError, KeyError) as err:
   1297             msg = "Unable to deserialize to object: " + class_name
-> 1298             raise_with_traceback(DeserializationError, msg, err)
   1299         else:
   1300             additional_properties = self._build_additional_properties(attributes, data)

c:\python36\lib\site-packages\msrest\exceptions.py in raise_with_traceback(exception, message, *args, **kwargs)
     49     error = exception(exc_msg, *args, **kwargs)
     50     try:
---> 51         raise error.with_traceback(exc_traceback)
     52     except AttributeError:
     53         error.__traceback__ = exc_traceback

c:\python36\lib\site-packages\msrest\serialization.py in _deserialize(self, target_obj, data)
   1292                         raw_value = found_value
   1293 
-> 1294                 value = self.deserialize_data(raw_value, attr_desc['type'])
   1295                 d_attrs[attr] = value
   1296         except (AttributeError, TypeError, KeyError) as err:

c:\python36\lib\site-packages\msrest\serialization.py in deserialize_data(self, data, data_type)
   1445             iter_type = data_type[0] + data_type[-1]
   1446             if iter_type in self.deserialize_type:
-> 1447                 return self.deserialize_type[iter_type](data, data_type[1:-1])
   1448 
   1449             obj_type = self.dependencies[data_type]

c:\python36\lib\site-packages\msrest\serialization.py in deserialize_iter(self, attr, iter_type)
   1476                 type(attr)
   1477             ))
-> 1478         return [self.deserialize_data(a, iter_type) for a in attr]
   1479 
   1480     def deserialize_dict(self, attr, dict_type):

c:\python36\lib\site-packages\msrest\serialization.py in <listcomp>(.0)
   1476                 type(attr)
   1477             ))
-> 1478         return [self.deserialize_data(a, iter_type) for a in attr]
   1479 
   1480     def deserialize_dict(self, attr, dict_type):

c:\python36\lib\site-packages\msrest\serialization.py in deserialize_data(self, data, data_type)
   1447                 return self.deserialize_type[iter_type](data, data_type[1:-1])
   1448 
-> 1449             obj_type = self.dependencies[data_type]
   1450             if issubclass(obj_type, Enum):
   1451                 if isinstance(data, ET.Element):

DeserializationError: Unable to deserialize to object: type, KeyError: 'DayOfWeek'
```

**Comment by tedchamb** [comment]
Fix is in #156

**Comment by tedchamb** [comment]
Fixed in [0.1.22](https://github.com/Microsoft/azure-devops-python-api/releases/tag/0.1.22)
